python/tests/graphql/test_nodes_property_filter.py

```diff
@@ -0,0 +1,792 @@
+import tempfile
+
+import pytest
+
+from raphtory.graphql import GraphServer
+from raphtory import Graph
+import json
+import re
+
+PORT = 1737
+
+
+def create_test_graph():
+    g = Graph()
+    g.add_node(1, "a", properties={"prop1": 60, "prop2": 31.3, "prop3": "abc123", "prop4": True, "prop5": [1, 2, 3]})
+    g.add_node(1, "b", properties={"prop1": 10, "prop2": 31.3, "prop3": "abc223", "prop4": False})
+    g.add_node(1, "c", properties={"prop1": 20, "prop2": 31.3, "prop3": "abc333", "prop4": True})
+    g.add_node(1, "d", properties={"prop1": 30, "prop2": 31.3, "prop3": "abc444", "prop4": False})
+    g.add_edge(
+        2,
+        "a",
+        "d",
+        properties={"eprop1": 60, "eprop2": 0.4, "eprop3": "xyz123", "eprop4": True, "eprop5": [1, 2, 3]},
+    )
+    g.add_edge(
+        2,
+        "b",
+        "d",
+        properties={"eprop1": 10, "eprop2": 1.7, "eprop3": "xyz123", "eprop4": True, "eprop5": [3, 4, 5]},
+    )
+    g.add_edge(
+        2,
+        "c",
+        "d",
+        properties={"eprop1": 30, "eprop2": 6.4, "eprop3": "xyz123", "eprop4": False, "eprop5": [10]},
+    )
+    return g
+
+
+def run_graphql_test(query, expected_output):
+    g = create_test_graph()
+    tmp_work_dir = tempfile.mkdtemp()
+    with GraphServer(tmp_work_dir).start(PORT) as server:
+        client = server.get_client()
+        client.send_graph(path="g", graph=g)
+
+        response = client.query(query)
+
+        # Convert response to a dictionary if needed and compare
+        response_dict = json.loads(response) if isinstance(response, str) else response
+        assert response_dict == expected_output
+
+
+def run_graphql_error_test(query, expected_error_message):
+    g = create_test_graph()
+    tmp_work_dir = tempfile.mkdtemp()
+    with GraphServer(tmp_work_dir).start(PORT) as server:
+        client = server.get_client()
+        client.send_graph(path="g", graph=g)
+
+        with pytest.raises(Exception) as excinfo:
+            client.query(query)
+
+        full_error_message = str(excinfo.value)
+        match = re.search(r'"message":"(.*?)"', full_error_message)
+        error_message = match.group(1) if match else ""
+
+        assert error_message == expected_error_message, f"Expected '{expected_error_message}', but got '{error_message}'"
+
+
+def test_node_property_filter_equal():
+    query = """
+    query {
+      graph(path: "g") {
+        nodes {
+            nodeFilter(
+              property: "prop5", 
+              condition: {
+                operator: EQUAL, 
+                value: [1, 2, 3]
```

Comment:
Can we get a test where prop 5 is on another node with [4,5,6] and see if this not returned